### ISSUE TITLE
Use unediting document for determining authLevels

### ISSUE THREAD

**Issue by makinde** [devcolor/mongoose-authz#15]
Right now a developer might do the following:

```js
const book = await Book.findOne().exec();
book.author = 'Someone New';
book.save({ ...some auth payload... });
```

The problem is that the document passed to `getAuthLevels` will be the updated document. You might end up allowing them to write to the object because the new value makes the current user the owner. This is problematic when exposing the model as a thinly wrapped REST API. You don't want someone to be able to take an arbitrary object ID, make themselves the owner of it, and then have get permission to do whatever they want on it.

I'm not sure the best way to do this. Some online forums suggest storing a copy of the unedited doc on `init`, which would work, but seems hacky. Others suggest using setters on attributes to store the old values. But this seems like it could get really messy since the schema may already have setters.

Another idea is  the when you are doing a non-lean query, that we forcibly turn on the permissions object so that gets stored with the doc. And then we configure that attribute to not be writable so it can't be changed. And then use that info to check permissions on save.

Any ideas?

**Comment by brysgo** [comment]
I think storing the results of getAuthLevel on load and using that when saving makes sense.

**Comment by makinde** [comment]
Fixed in #22
